packages/framework/attributor/src/mixinAttributor.ts

```diff
@@ -187,159 +100,20 @@ export const mixinAttributor = (
+					const result = await datastore.trySetAlias(attributorDataStoreAlias);
+					assert(result === "Success", "Failed to set alias for attributor data store");
+					runtimeAttributor = (await datastore.entryPoint.get()) as IRuntimeAttributor;
+					assert(runtimeAttributor !== undefined, "Attributor should be defined");
+				}
+			}
 
 			return runtime;
 		}
-
-		private runtimeAttributor: RuntimeAttributor | undefined;
-
-		protected addContainerStateToSummary(
-			summaryTree: ISummaryTreeWithStats,
-			fullTree: boolean,
-			trackState: boolean,
-			telemetryContext?: ITelemetryContext,
-		): void {
-			super.addContainerStateToSummary(summaryTree, fullTree, trackState, telemetryContext);
-			const attributorSummary = this.runtimeAttributor?.summarize();
-			if (attributorSummary) {
-				addSummarizeResultToSummary(summaryTree, attributorTreeName, attributorSummary);
-			}
-		}
 	} as unknown as typeof ContainerRuntime;
-
-class RuntimeAttributor implements IRuntimeAttributor {
-	public get IRuntimeAttributor(): IRuntimeAttributor {
-		return this;
-	}
-
-	public get(key: AttributionKey): AttributionInfo {
-		assert(
-			this.opAttributor !== undefined,
-			0x509 /* RuntimeAttributor must be initialized before getAttributionInfo can be called */,
-		);
-
-		if (key.type === "detached") {
-			throw new Error("Attribution of detached keys is not yet supported.");
-		}
-
-		if (key.type === "local") {
-			// Note: we can *almost* orchestrate this correctly with internal-only changes by looking up the current
-			// client id in the audience. However, for read->write client transition, the container might have not yet
-			// received a client id. This is left as a TODO as it might be more easily solved once the detached case
-			// is settled (e.g. if it's reasonable for the host to know the current user information at container
-			// creation time, we could just use that here as well).
-			throw new Error("Attribution of local keys is not yet supported.");
-		}
-
-		return this.opAttributor.getAttributionInfo(key.seq);
-	}
-
-	public has(key: AttributionKey): boolean {
-		if (key.type === "detached") {
-			return false;
-		}
-
-		if (key.type === "local") {
-			return false;
-		}
-
-		return this.opAttributor?.tryGetAttributionInfo(key.seq) !== undefined;
-	}
-
-	private encoder: Encoder<IAttributor, string> = {
-		encode: unreachableCase,
-		decode: unreachableCase,
-	};
-
-	private opAttributor: IAttributor | undefined;
-	public isEnabled = false;
-
-	public async initialize(
-		deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>,
-		quorum: IQuorumClients,
-		baseSnapshot: ISnapshotTree | undefined,
-		readBlob: (id: string) => Promise<ArrayBufferLike>,
-		shouldAddAttributorOnNewFile: boolean,
-	): Promise<void> {
-		const attributorTree = baseSnapshot?.trees[attributorTreeName];
-		// Existing documents that don't already have a snapshot containing runtime attribution info shouldn't
```

Comment:
Why don't we need this or similar back compat problem?  What was going on when this was added that has changed by now?

Reply:
Previously we were calling initialize here no matter what. Now we do this only if tracking is enabled. So, second case in if is gone. For the first case, the getRuntimeAttributor will just return undefined now as there is no attributor in snapshot. So, we can remove this.